opt_probs.py

```diff
@@ -275,7 +276,7 @@ def eval_node_probs(self):
         # Get probs
         probs = np.zeros([self.length, self.max_val, self.max_val])
 
-        if len(self.keep_sample) == 0:
+        if not self.keep_sample:
```

Comment:
This line causes the following error on my machine:
`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`
Could we revert back to the original version?
```suggestion
        if len(self.keep_sample) == 0:      
```